forge-game/src/main/java/forge/game/spellability/SpellAbilityRestriction.java

```diff
@@ -444,6 +444,11 @@ public final boolean checkOtherRestrictions(final Card c, final SpellAbility sa,
                 return false;
             }
         }
+        if (sa.isFreerunning()) {
```

Comment:
@tool4ever 
It might be better to make this more Generic? Same with Prowl?

Reply:
like scripting it into SVarToCheck directly in CardFactory so it's all in one place?
I wonder if there could be a risk of overwriting from collision but most likely it's safe 🤔

Reply:
For Freerunning we could make the Valid Hardcoded

for Prowl, we could change it to sharesCreatureTypeWith Spell Host

But yeah, we might do that later?